src/bp-core/bp-core-functions.php

```diff
@@ -8993,3 +8999,438 @@ function bb_load_reaction() {
+		array(
+			'object'   => $r['object'],
+			'item_id'  => $r['item_id'],
+			'text'     => $item_name,
+			'bg_color' => $all_palettes[ $palette ],
+		)
+	);
+
+	if ( 'user' === $r['object'] ) {
+		update_user_meta( $r['item_id'], 'default-user-avatar-png', $default_avatar );
+	} else {
+		groups_update_groupmeta( $r['item_id'], 'default-group-avatar-png', $default_avatar );
+	}
+
+	if ( ! empty( $default_avatar ) ) {
+		$prepare_response['url']  = $default_avatar;
+		$prepare_response['path'] = str_replace( bp_core_get_upload_dir( 'url' ), bp_core_avatar_upload_path(), $default_avatar );;
+	}
+
+	return $prepare_response;
+}
+
+/**
+ * Function to generate the default PNG avatar.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param array $args Array of avatar details.
+ *
+ * @return string
+ */
+function bb_generate_default_png_avatar( $args ) {
+	global $wp_filesystem;
+
+	$r = bp_parse_args(
+		$args,
+		array(
+			'item_id'    => 0,
+			'object'     => '',
+			'text'       => '',
+			'width'      => 300,
+			'height'     => 300,
+			'bg_color'   => '#008000',
+		)
+	);
+
+	if ( empty( $r['object'] ) || empty( $r['text'] ) || empty( $r['item_id'] ) ) {
+		return '';
+	}
+
+	/**
+	 * Set font family full path to render text on image.
+	 *
+	 * @since BuddyBoss [BBVSERION]
+	 *
+	 * @param string $font_family Full path of font family. It should be a TTF file.
+	 */
+	$font_family = apply_filters( 'bb_default_png_avatar_font_family', trailingslashit( buddypress()->plugin_dir ) . 'bp-core/fonts/SFUIText-Regular.ttf' );
+
+	if ( empty( $font_family ) ) {
+		return '';
+	}
+
+	/**
+	 * Set font color to render text on image.
+	 *
+	 * @since BuddyBoss [BBVSERION]
+	 *
+	 * @param string $png_text_color The color of the font to display on image.
+	 */
+	$png_text_color = apply_filters( 'bb_default_png_avatar_text_color', '#FFFFFF' );
+	if ( empty( $png_text_color ) ) {
+		$png_text_color = '#FFFFFF';
+	}
+
+	/**
+	 * Set font size to render text on image.
+	 *
+	 * @since BuddyBoss [BBVSERION]
+	 *
+	 * @param int $font_size The font size of the text to display on image.
+	 */
+	$font_size = (int) apply_filters( 'bb_default_png_avatar_text_font_size', 120 );
+	if ( empty( $font_size ) ) {
+		$font_size = 120;
+	}
+
+	// Generate filename.
+	$filename = time() . $r['item_id'] . '.png';
+
+	// Set upload directory and URL based on object.
+	$file_path = bp_core_avatar_upload_path() . '/group-avatars/default/' . $r['item_id'] . '/';
+	$file_url  = bp_core_get_upload_dir( 'url' ) . '/group-avatars/default/' . $r['item_id'] . '/' . $filename;
+	if ( 'user' === $r['object'] ) {
+		$file_path = bp_core_avatar_upload_path() . '/avatars/default/' . $r['item_id'] . '/';
+		$file_url  = bp_core_get_upload_dir( 'url' ) . '/avatars/default/' . $r['item_id'] . '/' . $filename;
+	}
+
+	if ( ! is_object( $wp_filesystem ) ) {
+		if ( ! function_exists( 'WP_Filesystem' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		WP_Filesystem();
+	}
```

Comment:
please create a function for this which can return the object $wp_filesystem and use whenever necessary.